python/cudf/cudf/core/series.py

```diff
@@ -2633,3 +2633,38 @@ def _align_indices(lhs, rhs, join="outer"):
         lhs = lhs_rhs.iloc[:, 0]
         rhs = lhs_rhs.iloc[:, 1]
     return lhs, rhs
+
+
+def _align_indices_multi(series_list):
+    """
+    Internal util to align the indices of a list of Series objects
+    """
+    head = series_list[0].index
+
+    all_index_equal = True
+    for sr in series_list[1:]:
+        if not sr.index.equals(head):
+            all_index_equal = False
+            break
+
+    if all_index_equal:
+        return series_list
+
+    aligned_index = cudf.core.reshape.concat(
+        [sr.index for sr in series_list]
+    ).unique()
+
+    result = []
+    for lhs in series_list:
+        rhs = Series(
+            column.column_empty(
+                len(aligned_index), dtype=lhs.dtype, masked=True
+            ),
+            index=aligned_index,
+        )
+        if len(lhs) != len(lhs.index.unique()) or len(rhs) != len(
+            rhs.index.unique()
+        ):
+            raise ValueError("Cannot align indices with non-unique values")
```

Comment:
We should check for uniqueness before doing the concatenation and empty allocations.